charts/kyverno-policies/templates/pod-sec-ctxt-spire.yaml

```diff
@@ -0,0 +1,41 @@
+{{- $name := "pod-sec-ctxt-spire" }}
+{{- include "kyverno-policy.supportedKyvernoCheck" (dict "top" . "ver" ">= 1.6.0-0") }}
+apiVersion: kyverno.io/v1
+kind: Policy
+metadata:
+  name: {{ $name }}
+  namespace: spire
+  annotations:
+    policies.kyverno.io/title: Require runAsNonRoot, run As Non-Root User, disallow Privilege Escalation and Privileged Containers
+    kyverno.io/kyverno-version: 1.6.0
+    policies.kyverno.io/subject: Pod
+    policies.kyverno.io/description: >-
+      Containers must be required to run as non-root users. Privilege escalation, such as via set-user-ID
+      or set-group-ID file mode, should not be allowed. Privileged mode disables most security mechanisms
+      and must not be allowed. This policy ensures `runAsNonRoot` is set to `true`, `runAsUser` and `runAsGroup`
+      is either unset or set to a number greater than zero, `allowPrivilegeEscalation` field is set to `false`
+      and pods do not call for privileged mode.
+spec:
+  rules:
+  - name: set-container-security-context
+    match:
+      resources:
+        kinds:
+        - Pod
+        selector:
+          matchLabels:
+            app.kubernetes.io/managed-by: Helm
+    mutate:
+      foreach:
+      - list: "request.object.spec.containers"
+        patchStrategicMerge:
+          spec:
+            containers:
+            - (name): |-
+                {{`{{element.name}}`}}
+              securityContext:
+                +(runAsUser): 65534

```

Comment:
We need to specify non-root  UID right?

Reply:
Is the workload already running with this UID? This is an optional set, if the workload doesn't already provide the setting. 

Since we're forcing runAsNonRoot, the workload will be forced to run with a non-UID 0 user. See https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#securitycontext-v1-core.

Reply:
So, shall we remove the UID setting in all the policies?